### DISCUSSION TITLE
Not able to update setting when using ACAP3 and axparameter

### DISCUSSION THREAD

**Discussion by DaveWomer** [AxisCommunications/discussions#143] [ACAP] [answered]
I'm trying to create a simple hello world app using ACAP3 and axparamter to update simple settings for my app.  I've created the paramConfig in the manifest.json and am able to see the parameters in the settings menu. I can also axis them from my application and change the settings programmatically.  However, when I attempt to change the setting from the camera web UI I get an error stating "Oops, Couldn't set the value".   It doesn't seem to matter of the app is running or not.  I don't see a resulting error in my log.  Any help is appreciated!

**Comment by mattias-kindborg-at-work** [comment]
The [OPC UA Server ACAP](https://github.com/AxisCommunications/opc-ua-server-acap) application on GitHub is using parameters. Please see the code for an example on how to define a parameter.

**Comment by daniel-falk** [comment, marked as answer]
Hi @DaveWomer,

I have a simple example of using the parameter handler on my github, see link below. There are some things that can cause the parameter library to fail, e.g. if the application name or the parameter name contains other characters than `a-z`, `A-Z` or `0-9`, then the parameters will fail to update.

If you do not need to use callbacks on parameter update, then you do not need to have a glib event loop running, so the example gets a bit easier. If my example does not work for you, please tell me and I will look at it. You can find it here:
https://github.com/daniel-falk/Axis-ACAP-guides/tree/main/parameterhandler/SDK3

**Reply by nixpanic** [reply]
oh, yes, if an `appName` contains a `_`  like `hello_world`, the AXIS webui does not allow setting the parameters, but it works through the API or with the `parhandclient` command.
